lua/telescope/_extensions/project.lua

```diff
@@ -0,0 +1,62 @@
+local has_telescope, telescope = pcall(require, 'telescope')
+
+if not has_telescope then
+	error('This plugins requires nvim-telescope/telescope.nvim')
+end
+
+local actions = require("telescope.actions")
+local finders = require("telescope.finders")
+local pickers = require("telescope.pickers")
+local conf = require("telescope.config").values
+local builtin = require("telescope.builtin")
+local project_dirs_file = vim.fn.stdpath('data') .. '/telescope-projects.lua'
+
+local select_project = function(opts, projects, run_task_on_selected_project)
+	pickers.new(opts, {
+		prompt_title = 'Select a project',
+		results_title = 'Projects',
+		finder = finders.new_table {results = projects},
+		sorter = conf.file_sorter(opts),
+		attach_mappings = function(prompt_bufnr)
+			local on_project_selected = function()
+				local selection = actions.get_selected_entry(prompt_bufnr)
+				actions.close(prompt_bufnr)
+				run_task_on_selected_project(selection.value)
+			end
+			actions.goto_file_selection_edit:replace(on_project_selected)
+			return true
+		end
+	}):find()
+end
+
+local project = function(opts)
+	opts = opts or {}
+
+	-- TODO figure this out:
+
+	-- local project_dirs = {}
+	-- for line in io.lines(project_dirs_file) do
+	-- 	table.insert(project_dirs, line)
+	-- end
+
+	local project_dirs = {["nvim config"] = "~/.config/nvim"}
+	local projects = {}
+
+	local search_selected_project = function(selection)
```

Comment:
selection will now be the table that you inserted to projects.
So take a look at it with `print(vim.inspect(selection))` to see how to extract the path